viewport/index.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+import { reduce, forEach, debounce, mapValues, property } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { dispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import './store';
+
+export { default as ifViewportMatches } from './if-viewport-matches';
+
+/**
+ * Hash of breakpoint names with pixel width at which it becomes effective.
+ *
+ * @see _breakpoints.scss
+ *
+ * @type {Object}
+ */
+const BREAKPOINTS = {
```

Comment:
can we do something similar to what gutenberg/edit-post/store/constants.js does? Use the scss variable loader '!!sass-variables-loader!../assets/stylesheets/_breakpoints.scss';

Reply:
>can we do something similar to what gutenberg/edit-post/store/constants.js does? Use the scss variable loader '!!sass-variables-loader!../assets/stylesheets/_breakpoints.scss';

My main issue with this is that the breakpoints are defined in an `edit-post` stylesheet. Ideally those common SCSS variables don't remain there. The main issue we solve with the variable loader is avoiding repetition, but one of the goals with viewport module is at least to consolidate all viewport-related logic. So _hopefully_ at most we'd have at most two sets to maintain: one in SCSS and one in JS.

Reply:
Yes it's trade that makes sense to make, and as it is just this cases, it is not hard to maintain 👍